SamlValve/src/main/java/com/axonivy/demo/saml/UserAuthenticator.java

```diff
@@ -0,0 +1,205 @@
+package com.axonivy.demo.saml;
+
+import java.io.ByteArrayInputStream;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+import java.util.Base64;
+import java.util.Optional;
+
+import javax.xml.parsers.DocumentBuilderFactory;
+
+import org.apache.catalina.connector.Request;
+import org.apache.http.auth.AuthenticationException;
+import org.joda.time.DateTime;
+import org.opensaml.saml.saml2.core.Assertion;
+import org.opensaml.saml.saml2.core.Conditions;
+import org.opensaml.saml.saml2.core.Response;
+import org.opensaml.security.credential.Credential;
+import org.opensaml.security.x509.BasicX509Credential;
+import org.opensaml.xml.Configuration;
+import org.opensaml.xml.io.Unmarshaller;
+import org.opensaml.xml.io.UnmarshallerFactory;
+import org.opensaml.xml.io.UnmarshallingException;
+import org.opensaml.xml.parse.BasicParserPool;
+import org.opensaml.xml.parse.XMLParserException;
+import org.opensaml.xmlsec.signature.support.SignatureException;
+import org.opensaml.xmlsec.signature.support.SignatureValidator;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+
+/**
+ * This class is responsible for authenticating a user given in a SAML Response.
+ */
+public final class UserAuthenticator
+{
+  private static final String REQUEST_SAML_PARAMETER_NAME = "SAMLResponse";
+
+  // This needs to be changed to the real path to the certificate
+  private static final X509Certificate SIGN_VERIFICATION_CERT = loadCertificate("myCertificate");

```

Comment:
hello @ivy-arus 
because of this, the valve can not work by default without further modifications. I would:
1. Remove the signature check
or 
2. Make the certifacte file configurable and if it does not exist, do not check it.